harness/tests/test_workload_manager.py

```diff
@@ -0,0 +1,107 @@
+import contextlib
```

Comment:
praise: tests!

We talked a bit about moving this into a test utility file offline, decided this stuff is all workload manager specific for the most part.